Task BJS2-68061

MightySonnya opened the pull request:
1) Class User added
2) Method findHobbyLovers added

MightySonnya committed:
Task BJS2-68061

1) Class User added
2) Method findHobbyLovers added

MarselYsup commented on src/main/java/school/faang/hobbyes/User.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Override
+    public String toString() {
```
toString можно сгенерировать при помощи Lombok аннотации ```@ToString```, так твой код будет более гибким, если добавится новое поле

MarselYsup commented on src/main/java/school/faang/hobbyes/User.java:
```diff
@@ -0,0 +1,37 @@
+public class User {
+    private String name;
+    private int age;
+    private Set<String> manyActions;
```
как будто бы ```hobbies``` или ```activities``` звучит более понятно

MarselYsup commented on src/main/java/school/faang/hobbyes/User.java:
```diff
@@ -0,0 +1,37 @@
+        Map<User, String> result = new HashMap<>();
+        for (User user : users) {
+            Set<String> userHobbies = user.getManyActions();
+            for (String hobby : userHobbies) {
```
можно сразу в forEach не создавая лишней ссылки сверху, 
вот так
```java
for (String hobby : user.getManyActions()) {
```

MarselYsup commented on src/main/java/school/faang/hobbyes/User.java:
```diff
@@ -0,0 +1,37 @@
+    private String name;
+    private int age;
+    private Set<String> manyActions;
+    private int id;
```
по опыту советую поле id всегда писать первым из полей, так как оно является одним из определяющим для пользователя + когда пойдет Hibernate на id обычно несколько аннотаций накладывают и становиться неудобно когда это поле где то в середине полей

MightySonnya committed:
Task BJS2-68061

1) put Id on right place
2) rework for -> forEach
3) generate annotation
4) Rename Hobbyes
